src/renderer/file-manager.ts

```diff
@@ -68,7 +69,39 @@ export class FileManager {
     if (!filePath || typeof filePath !== 'string') return;
 
     const editorValues = {};
-    for (const [name, value] of Object.entries(await readFiddle(filePath))) {
+    const files: [string, string][] = Object.entries(
+      await readFiddle(filePath),
+    );
+    for (const [name, value] of files) {
+      if (name === PACKAGE_NAME) {
+        const { remoteLoader } = window.ElectronFiddle.app;
+        const { dependencies, devDependencies } = JSON.parse(value);
+        const deps: Record<string, string> = {
+          ...dependencies,
+          ...devDependencies,
+        };
+
+        if (deps.electron) {
+          // Strip off semver range prefixes.
+          const index = deps.electron.search(/\d/);
+          const version = deps.electron.substring(index);
+
+          if (!semver.valid(version)) {
+            throw new Error(
+              "This Fiddle's package.json contains an invalid Electron version.",
+            );
+          }
+
+          remoteLoader.setElectronVersion(version);
+
+          // We want to include all dependencies except Electron.
+          delete deps.electron;
+        }
+
+        this.appState.modules = new Map(Object.entries(deps));
+        continue;
+      }
+
```

Comment:
Should this new code go in `app.replaceFiddle()` somewhere so that other codepaths that replace the fiddle will also get these Electron version safety checks?

Reply:
Would you potentially be ok with taking that as a follow-up? i'd want to think about the best way to do that but agree we should extract this to some extent 🤔

Reply:
I wouldn't block this PR on it but I would want to ensure we commit to a followup. Otherwise we'll have multiple codepaths that do almost-but-not-quite the same thing, by accident rather than by design, which is a pain to debug & maintain.